### PR TITLE
Add '@rollup/plugin-esm-shim'

### PR DESCRIPTION
**Pull request by Peeterush**
A new attempt at getting both ESM and UMD builds to play well!

### CODE REVIEW

**Comment by Peeterush** [comment]
Ah, Deno is like that one dentist.

I made a [PR](https://github.com/rollup/plugins/pull/1622) to `@rollup/plugin-esm-shim` to add the prefix that Deno is asking for.

**Comment by Peeterush** [comment]
Edit: added a test for dist files like [suggested by iambumblehead](https://github.com/dlemstra/magick-wasm/discussions/119#discussioncomment-7474264).

File: tests/dist/test-ESM.mjs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// Copyright Dirk Lemstra https://github.com/dlemstra/magick-wasm.
+// Licensed under the Apache License, Version 2.0.
+
+import { initializeImageMagick, Magick } from '../../dist/index.mjs';
+
+const wasmLocation = '../../node_modules/@dlemstra/magick-native/magick.wasm';
+
+initializeImageMagick(wasmLocation).then(() => {
+  console.log(Magick.imageMagickVersion);
```
Comment (dlemstra) [inline tests/dist/test-ESM.mjs:9]: We could also use the `features` here instead so we don't need a regex?

**Comment by dlemstra** [comment]
Can we avoid using that module if `magick-native` would provide a `magick.js` and `magick.mjs` file? There is a build option in emscripten sdk to create a ES6 module. How should I package those files in a package.json file?

File: tests/dist/test-ESM.mjs (hunk 1 above)
Comment (Peeterush) [reply tests/dist/test-ESM.mjs]: Good idea!

**Comment by Peeterush** [comment]
ESM would be better yes! Vite would only use the ESM if it's available, so no CJS version needed. Package should look something like this:
```json
{
    "name": "@dlemstra/magick-native",
    "version": "",
    "type": "module",
    "exports": {
        ".": {
            "types": "./index.d.ts",
            "default": "./index.js",
        },
    },
    "module": "./index.js",
    "types": "./index.d.ts",
    "repository": {
        "type": "git",
        "url": "git+https://github.com/dlemstra/Magick.Native.git",
    },
    "author": "Dirk Lemstra",
    "license": "Apache-2.0",
    "files": [
        "./index.d.ts",
        "./index.js",
        "magick.wasm",
        "NOTICE",
    ],
}
```
Note: because of `"type": "module"`, any `.js` is assumed to be ESM, so no `.mjs` needed.

**Comment by Peeterush** [comment]
Also; my PR to `@rollup/plugin-esm-shim` has been accepted and published, so Deno is happy now!

**Comment by dlemstra** [comment]
Do we still need the `@rollup/plugin-esm-shim` once `Magick.Native` has switched to ESM?

**Comment by dlemstra** [comment]
I have made some changes to the Magick.Native build but I did not rename the files. Can you check if that works?

**Pull request merged**